docs/guides/6to7guide.md

```diff
@@ -0,0 +1,157 @@
+---
+title: restify 6.x to 7.x migration guide
+permalink: /docs/6to7/
+---
+
+## Introduction
+
+restify `7.x` comes with a completely new router and middleware logic that
+brings significant performance improvement to your application.
+From `v7.0.0` restify uses the Radix Tree based
+[find-my-way](https://github.com/delvedor/find-my-way) package as a router
+backend.
+
+## Breaking Changes
+
+### Server throws `RequestCloseError` instead of `RequestAbortedError`
+
+Server throws `RequestCloseError` instead of `RequestAbortedError` in the case
+of the request was terminated by the client for some reason.
+
+The new version of restify never throws `RequestAbortedError`.
+
+### Non-strict routing is gone
+
+Option `strictRouting` is removed `createServer({ strictRouting: false })`.
+Strict routing is the new default.
+
+### Different `RegExp` usage in router path and wildcards
+
+restify's new router backend
+[find-my-way](https://github.com/delvedor/find-my-way) has limited RegExp
+support.
+
+#### Guide to define paths
+
+To register a **parametric** path, use the *colon* before the parameter name.
+For **wildcard** use the *star*.
+*Remember that static routes are always inserted before parametric and wildcard.*
+
+```js
+// parametric
+server.get('GET', '/example/:userId', (req, res, next) => {}))
+server.get('GET', '/example/:userId/:secretToken', (req, res, next) => {}))
+
+// wildcard
+server.get('GET', '/example/*', (req, res, next) => {}))
+```
+
+Regular expression routes are supported as well, but pay attention, RegExp are
+very expensive in term of performance!
+
+```js
+// parametric with RegExp
+server.get('GET', '/example/:file(^\\d+).png', () => {}))
+```
+
+RegExp path chunks needs to be between parentheses.
+
+It's possible to define more than one parameter within the same couple of slash
+("/"). Such as:
+
+```js
+server.get('/example/near/:lat-:lng/radius/:r', (req, res, next) => {}))
+```
+
+*Remember in this case to use the dash ("-") as parameters separator.*
+
+Finally it's possible to have multiple parameters with RegExp.
+
+```js
+server.get('/example/at/:hour(^\\d{2})h:minute(^\\d{2})m', (req, res, next) => {
+  // req.params => { hour: 12, minute: 15 }
+}))
+```
+In this case as parameter separator it's possible to use whatever character is
+not matched by the regular expression.
+
+Having a route with multiple parameters may affect negatively the performance,
+so prefer single parameter approach whenever possible, especially on routes
+which are on the hot path of your application.
+
+Fore more info see: https://github.com/delvedor/find-my-way
+
+### Remove already deprecated `next.ifError`
+
+`next.ifError(err)` is not available anymore.
+
+### Disable DTrace probes by default
+
+DTrace probes comes with some performance impact that's fine for the sake of
+observability but you may don't use it at all.
+
+### Change in calling `next` multiple times
```

Comment:
Is performance the motivation behind this new default behavior?

Reply:
Also have a preference for this to remain unchanged. Throwing by default helps us catch a whole class of bugs.

Reply:
Yes, performance is the motivation behind.
@retrohacker do you mean at Netflix? We can re-enable it with the option for NQ services.

Reply:
@hekike I just mean in general. We are a framework for building production ready REST services, I'm not sure this is in line with our goals as a project.